src/window/src/lib.rs

```diff
@@ -21,54 +35,93 @@ pub struct Size {
     pub height: u32,
 }
 
+/// Convinience conversion between an array of size two, and a size structure.
 impl From<[u32; 2]> for Size {
     #[inline(always)]
     fn from(value: [u32; 2]) -> Size {
         Size { width: value[0], height: value[1] }
     }
 }
 
+/// Convinience conversion between a tuple of (width,height), and a size structure.
 impl From<(u32, u32)> for Size {
     #[inline(always)]
     fn from(value: (u32, u32)) -> Size {
         Size { width: value.0, height: value.1 }
     }
 }
 
-/// Builds window from window settings.
+/// This trait indicates that a window can be built from a WindowSettings object.
 pub trait BuildFromWindowSettings: Sized {
-    /// Builds window from window settings.
+    /// Builds the window from a window settings object.
+    ///
+    /// # Errors
+    ///
+    /// See your backend's documentation for details about what kind of
+    /// error strings can be returned, and the conditions for error.
     fn build_from_window_settings(settings: WindowSettings)
     -> Result<Self, String>;
 }
 
-/// Required to use the event loop.
+/// Trait for piston internals to interface with a window backend.
+///
+/// This trait defines all the behavior needed for making an event loop.
+/// Event loop definitions can be found in [`piston::event_loop`]
+/// (../event_loop/index.html).
+/// An example of a working event loop can be found in the Piston-Tutorials
+/// repository, under getting-started, or in the event loop examples.
+/// The window trait methods are private; they're mostly for use in the abstract
+/// code in the `event_loop` module.
 pub trait Window {
-    /// The event type emitted by `poll_event`
+    /// The event type the window uses for incoming input.
+    /// 
+    /// Usually, this will be event_loop::Input, but may vary
+    /// between implementations if more or less information is available.
+    ///
+    /// For example, if a backend doesn't support mouse input because it's designed
+    /// to be sent over a network, then it might use a different event type. 
     type Event;
 
-    /// Tells the window to close or stay open.
+    /// Tells the window to close or stay open. 
+    ///
+    /// Usually paired with a public method in the specific window implementation.
     fn set_should_close(&mut self, value: bool);
 
     /// Returns true if window should close.
+    ///
+    /// Usually paired with a public method in the specific window implementation.
     fn should_close(&self) -> bool;
 
     /// Gets the size of the window in user coordinates.
+    ///
+    /// Usually paired with a public method in the specific window implementation.
     fn size(&self) -> Size;
 
     /// Swaps render buffers.
+    ///
+    /// Client code shouldn't ever have to deal with this.
     fn swap_buffers(&mut self);
 
     /// Polls event from window.
+    ///
+    /// To read events in client code, look at the
```

Comment:
"application code"

"client" might be misleading to think of a client/server interface.